packages/ui-components/src/Badge/Badge.tsx

```diff
@@ -36,7 +36,7 @@ export const Badge: FunctionComponent<BadgeProps> = ({
 
   if (children !== undefined) {
     return (
-      <div className={classnames} {...props}>
+      <div data-testid="badge" className={classnames} {...props}>
```

Comment:
why is this data attribute necessary?

Reply:
since we can't get a div using `getByRole`. Introduced a `data-testid` to get the element.

Reply:
Oh! I see the use of `screen.getByTestId`.

Reply:
`data-testid` is really useful to test elements that are deep inside the dom tree and we don't have to do much work.

Reply:
I would prefer not to add data attributes to the component. Could we instead use `screen.getByText` to do something like

```js
  it("should able to change the intent type", () => {
    const text = "intent test";
    const { rerender } = render(<Badge>{text}</Badge>);

    // success
    rerender(<Badge intent="success">{text}</Badge>);
    expect(screen.getByText(text)).toHaveClass("intent-success");

// etc etc etc
```

Reply:
sure, will update it.

Reply:
Thanks!

Reply:
updated 👍🏻